### PR TITLE
Commodore64x - /glist improvement

### PR DESCRIPTION
**Pull request by ghost**
Completely restructure command - removed unneeded or unused code as well as adding in glist banip, unbanip (supports range and normal ips), nameban and unbanname.

### CODE REVIEW

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_glist.java [hunk 1]
```diff
@@ -12,104 +12,164 @@
...
 
     @Override
     public boolean run(CommandSender sender, Player playerSender, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
     {
-        if (args.length < 1)
-        {
-            return false;
-        }
-
-        if (args.length == 1)
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_glist.java:27]: You've removed this however still listed it as a potential argument.

**Review by Wild1145 — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
@Wild1145 All issues have been rectified

**Comment by lemonsked** [comment]
Why  remove purge?

**Comment by Wild1145** [comment]
Thank you for contributing to the project, could you please amend your opening message to include the issue that this PR resolves. Thank You.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_glist.java [hunk 2]
```diff
@@ -12,104 +12,164 @@
...
+                if (player != null)
+                {
+                    player.kickPlayer(playerBan.bakeKickMessage());
+                }
+                return true;
+            case "unban":
+            case "pardon":
+                if (usingIp)
+                {
+                    msg("Please specify a player, not an ip.");
```
Comment (LiamTownsley) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_glist.java:88]: You could make it so you can /glist unban [USERNAME/IP]

**Comment by mayokek** [comment]
`/glist purge` is actually useful as `/gtfo` isn't made for 24-48 hours ban, it permanent ban them.

**Review by Wild1145 — CHANGES_REQUESTED** [review]
The functionality that purge offers is something that we would like to remain in the project.

**Comment by ghost** [comment]
@Wild1145 Purge is not really ever used and causes issues since it doesn't just purge ban data.

**Comment by Wild1145** [comment]
@Commodore64x I understand it is fairly regularly used, and if it is breaking /cage when we need to fix /cage. I thought I had done that with a recent PR for cage improvements anyway but that might not be the case.

**Comment by ghost** [comment]
@Wild1145 Searching logs from the last 10 days shows noone has used it once, it really isn't practical to have a command that clears all fplayer and playerdata data as it does break a lot of things.

**Comment by Wild1145** [comment]
@Commodore64x Can you elaborate on what exactly it breaks?

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_glist.java [hunk 3]
```diff
@@ -12,104 +12,164 @@
...
+                plugin.bm.addBan(nameBan);
 
-            for (String ip : ips)
-            {
-                Ban ban = plugin.bm.getByIp(ip);
-                if (ban != null)
+                if (player != null)
                 {
-                    plugin.bm.removeBan(ban);
+                    player.kickPlayer(nameBan.bakeKickMessage());
```
Comment (LiamTownsley) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_glist.java:123]: `.bakeKicKMessage()`?

**Review by LiamTownsley — CHANGES_REQUESTED** [review]

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_glist.java (hunk 3 above)
Comment (lemonsked) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_glist.java:123]: It's the same as before.

**Comment by ghost** [comment]
@Wild1145 It simply isn't ever used. It clears fPlayer and player data which is far from ideal. Things like muting, blocked commands, cages, freezing has issues and the /purgeall command is favored as a safer and easier way to achieve the unfreezing/caging etc of players.

**Comment by mayokek** [comment]
And how does the banning affect the player data?

**Comment by ghost** [comment]
@mayokek It doesn't, however /glist purge does.

**Comment by mayokek** [comment]
@Commodore64x, `/glist purge` does only purge banlist...

**Comment by mayokek** [comment]
@Commodore64x, I see the issue, it purges the player data, not banlist. Might wanna change `plugin.pl.purge();` to `plugin.bm.purge();` if you are going to add it back.

**Comment by Wild1145** [comment]
@Commodore64x We do not want to be removing functionality for shitts and giggles just because you dont see a use case for it. The issue stands that you are simply removing functionality because you feel like it, without any associated issue for discussion. Until an approved issue is raised to remove the /glist purge functionality, it is to remain in TFM.

**Pull request closed without merging**